File: Python/title_to_number.py
# Given a column title as appear in an Excel sheet, return its corresponding column number.

def titleToNumber(s):
    length = len(s)
    sum = 0

    for i in range(0, length):
        sum += (ord(s[i]) - 64) * 26 ** (length - i - 1)

    return sum


# print(titleToNumber('A'))  # 1
# print(titleToNumber('AB'))  # 28
# print(titleToNumber('ZY'))  # 701
# print(titleToNumber('AAA'))  # 703

File: Javascript/titleToNumber.js
// Given a column title as appear in an Excel sheet, return its corresponding column number.

const titleToNumber = (s) => {
  let sum = 0;
  const len = s.length;

  for (let i = 0; i < len; i++) {
    const letterCode = (s[i].charCodeAt(0) - 64) * 26 ** (len - i - 1);
    sum += letterCode;
  }

  return sum;
};

// console.log(titleToNumber("A")); // -> 1
// console.log(titleToNumber("AB")); // -> 28
// console.log(titleToNumber("ZY")); // -> 701
// console.log(titleToNumber("AAA")); // -> 703
